MyNumSetLibrary/MyNumSetLibrary.Tests/NumSetTests.cs

```diff
@@ -0,0 +1,239 @@
+
+            //Assert
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void NumSet_ToString_ThrowsArgumentNullException()
+        {
+            //Assign
+            NumSet numSet = new(null!);
+
+            //Act
+            numSet.ToString();
+
+            //Assert
+        }
+
+        [TestMethod]
+        [DataRow(new int[] { }, "")]
+        [DataRow(new int[] { 1, 2, 3, 4, 5 }, "1,2,3,4,5")]
+        public void NumSet_ToString(int[] array, string expected)
+        {
+            //Assign
+            NumSet numSet = new(array!);
+
+            //Act
+
+            //Assert
+            Assert.AreEqual(expected, numSet.ToString());
+        }
+
+        [TestMethod]
+        public void NumSet_Equals_FalseWithNull()
+        {
+            //Assign
+            int[] array = Array.Empty<int>();
+            NumSet left = new(array);
+
+            //Act
+
+            //Assert
+            Assert.IsFalse(left.Equals(null));
+        }
+
+        [TestMethod]
+        [DataRow(new int[] { }, new int[] { 1 })]
+        [DataRow(new int[] { 1 }, new int[] { })]
+        public void NumSet_Equals_FalseWithValue(int[] leftArray, int[] rightArray)
+        {
+            //Assign
+            NumSet left = new(leftArray);
+            NumSet right = new(rightArray);
+
+            //Act
+
+            //Assert
+            Assert.IsFalse(left.Equals(right));
+        }
+
+        [TestMethod]
+        public void NumSet_Equals_FalseDifferentObject()
+        {
+            //Assign
+            int[] array = new int[] { 1 };
+            NumSet left = new(array);
+            HashSet<int> hasSet = new(array);
+
+            //Act
+
+            //Assert
+            Assert.IsFalse(left.Equals(hasSet));
+        }
+
+        [TestMethod]
+        public void NumSet_Equals_TrueWithValue()
+        {
+            //Assign
+            int[] array = new int[] { 1 };
+            NumSet left = new(array);
+            NumSet right = new(array);
+
+            //Act
+
+            //Assert
+            Assert.IsTrue(left.Equals(right));
+        }
+
+        [TestMethod]
+        public void NumSet_Equals_TrueSameObject()
+        {
+            //Assign
+            NumSet left = new(1);
+            NumSet right = left;
+
+            //Act
+
+            //Assert
+            Assert.IsTrue(left.Equals(right));
+        }
+
+        [TestMethod]
+        [DataRow(new int[] { }, 0)]
+        [DataRow(new int[] { 0 }, 1)]
+        [DataRow(new int[] { 1 }, 2)]
+        [DataRow(new int[] { 1, 2 }, 5)]
+        [DataRow(new int[] { 5, 3, 1, 4, 2 }, 20)]
+        public void NumSet_GetHashCode(int[] array, int hashCode)
+        {
+            //Assign
+            NumSet numSet = new(array);
+
+            //Act
+
+            //Assert
+            Assert.AreEqual(hashCode, numSet.GetHashCode());
+        }
+
+        [TestMethod]
+        public void NumSet_EqualOperator_False()
```

Comment:
One technique that is often used to reduce the number of tests we need to write with equality operators, is to test the operator at the same time you test the Equals methods since it will be the same set of inputs and expectations.

Reply:
That makes sense.